server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java

```diff
@@ -374,12 +377,17 @@ public void upload(MultipartFile dsFile,
         String jobContent = "";
         try (final InputStream inputStream = dsFile.getInputStream()) {
             // only extract the eval job file content
+            // TODO: replace with oss path content
+            // but update only for job
             jobContent = new String(
                     Objects.requireNonNull(
                             TarFileUtil.getContentFromTarFile(dsFile.getInputStream(), "src", "eval_jobs.yaml")));
-            storageAccessService.put(String.format(FORMATTER_STORAGE_PATH, modelPath, dsFile.getOriginalFilename()),
-                    inputStream, dsFile.getSize());
-        } catch (IOException e) {
+            TarFileUtil.extract(inputStream, (name, size, in) ->
```

Comment:
since client uploading non-archived files, should this block be deleted?

Reply:
new files contain: manifest.yaml/model files/code files(tar),so code files also need to be extracted